### DISCUSSION TITLE
changes configure_file input rebuilt the buildir like meson setup build --reconfigure

### DISCUSSION THREAD

**Discussion by aullidolunar** [mesonbuild/meson#13654] [Q&A] [answered]
Basically:
```
blp_file = configure_file(
	input: data_sources.get(0), # builder.blp.in
	output: data_files.get(0).substring(0, -3), # builder.blp
	configuration: {
		'PROGRAM_NAME': PROGRAM_NAME,
		'PROGRAM_TITLE': PROGRAM_TITLE
	}
)
```
Any changes in builder.blp.in causes meson to reconfigure all over, any ideas about how to only update the configure_file similar when you make changes in sources files to compile only the sources changed?

**Comment by eli-schwartz** [comment, marked as answer]
This happens because configure_file itself runs at configure time, whereas a custom_target runs at build time. Usually, configured files tend to be files that very rarely change.

It's possible to instead use a `custom_target()` where the command is something like sed. Or perhaps a custom python script that performs the necessary replacements. You'd pass PROGRAM+NAME and PROGRAM_TITLE as arguments to your sed/python script.

**Reply by aullidolunar** [reply]
Thanks for the input.
